operator/pkg/resources/statefulset.go

```diff
@@ -1024,58 +1025,41 @@ func (r *StatefulSetResource) AdditionalListenersEnvVars() []corev1.EnvVar {
 // - pandaproxy.pandaproxy_api
 // - pandaproxy.advertised_pandaproxy_api
 // example: redpanda.kafka_api: "[{'name':'private-link','address':'0.0.0.0','port':39002}]"
-// example: redpanda.advertised_kafka_api: "[{'name':'private-link','address':'{{ .Index }}-f415bda0-{{ .HostIP | sha256sum | substr 0 7 }}.cluster123.fmc.prd.cloud.redpanda.com','port': 'port': {{30092 | add .Index}}}]"
 func (r *StatefulSetResource) GetPortsForListenersInAdditionalConfig() []corev1.ContainerPort {
 	ports := []corev1.ContainerPort{}
 
 	if len(r.pandaCluster.Spec.AdditionalConfiguration) == 0 {
 		return ports
 	}
 
-	additionalNode0Config := config.ProdDefault()
-	for _, k := range additionalListenerCfgNames {
-		if v, found := r.pandaCluster.Spec.AdditionalConfiguration[k]; found {
-			res, err := utils.Compute(v, utils.NewEndpointTemplateData(0, "dummy", 0), false)
-			if err != nil {
-				r.logger.Error(err, "failed to evaluate template", "template", v)
-				continue
-			}
-			err = config.Set(additionalNode0Config, k, res)
+	getPorts := func(key string) []corev1.ContainerPort {
+		if v, found := r.pandaCluster.Spec.AdditionalConfiguration[key]; found {
+			var ports []corev1.ContainerPort
+			res, err := utils.Compute(v, utils.NewEndpointTemplateData(0, "dummy", r.nodePool.HostIndexOffset), false)
 			if err != nil {
-				r.logger.Error(err, "failed to set node config", k, v)
-				continue
+				return nil
 			}
-		}
-	}
 
-	for i := 0; i < int(ptr.Deref(r.nodePool.Replicas, 0)); i++ {
-		for _, n := range additionalNode0Config.Redpanda.AdvertisedKafkaAPI {
-			ports = append(ports, corev1.ContainerPort{
-				Name:          getAdditionalListenerPortName(n.Name, i),
-				ContainerPort: int32(n.Port + i),
-			})
-		}
-		if additionalNode0Config.Pandaproxy != nil {
-			for _, n := range additionalNode0Config.Pandaproxy.AdvertisedPandaproxyAPI {
+			var kafkaAPI []config.NamedAuthNSocketAddress
```

Comment:
nit: replace `kafkaAPI` with e.g. `addrs` since `getPorts()` is not just for Kafka API.

Reply:
Done